hudi-common/src/main/java/org/apache/hudi/common/util/HoodieAvroUtils.java

```diff
@@ -214,6 +220,63 @@ private static GenericRecord rewrite(GenericRecord record, Schema schemaWithFiel
     return newRecord;
   }
 
+  /*
+  This function takes the union of all the fields except hoodie metadata fields
+   */
+  private static List<Field> getAllFieldsToWrite(Schema oldSchema, Schema newSchema) {
+    Set<Field> allFields = new HashSet<>(oldSchema.getFields());
+    List<Field> fields = new ArrayList<>(oldSchema.getFields());
+    for (Schema.Field f : newSchema.getFields()) {
+      if (!allFields.contains(f) && !isMetadataField(f.name())) {
+        fields.add(f);
+      }
+    }
+
+    return fields;
+  }
+
+  private static void populateNewRecordAsPerDataType(GenericRecord record, Field field) {
+    switch (getSchemaTypeForField(field)) {
+      case STRING:
+      case BYTES:
+      case ENUM:
+      case FIXED:
+        record.put(field.name(), field.defaultVal() == null ? null : (String) field.defaultVal());
```

Comment:
Why do we need this casting of individual data types ? It seems we can just pass `field.defaultVal()` as it is because it expects an `Object`, and `field.defaultVal()` returns exactly that.

Reply:
Done.